core/os/os.cpp

```diff
@@ -359,7 +362,46 @@ String OS::get_unique_id() const {
 }
 
 int OS::get_processor_count() const {
+#if defined(_WIN32) || defined(_WIN64)
+	DWORD length = 0;
+	int concurrency = 0;
+	const auto validConcurrency = [&concurrency]() noexcept -> int {
+		if (concurrency == 0)
+			return std::thread::hardware_concurrency();
+		else
+			return concurrency;
+	};
+	if (GetLogicalProcessorInformationEx(RelationAll, nullptr, &length) != FALSE) {
+		return validConcurrency();
+	}
+	if (GetLastError() != ERROR_INSUFFICIENT_BUFFER) {
+		return validConcurrency();
+	}
+	std::unique_ptr<void, void (*)(void *)> buffer(std::malloc(length), std::free);
+	if (!buffer) {
+		return validConcurrency();
+	}
+	auto *mem = reinterpret_cast<unsigned char *>(buffer.get());
+	if (GetLogicalProcessorInformationEx(
+				RelationAll, reinterpret_cast<PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX>(mem), &length) == false) {
+		return validConcurrency();
+	}
+	DWORD i = 0;
+	while (i < length) {
+		const auto *proc = reinterpret_cast<PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX>(mem + i);
```

Comment:
`auto` is [not allowed](https://docs.godotengine.org/en/stable/contributing/development/cpp_usage_guidelines.html#auto-keyword)